controls/nginx_spec.rb

```diff
@@ -181,3 +194,64 @@
     its('add_header') { should include 'X-Content-Type-Options nosniff' }
   end
 end
+
+control 'nginx-11' do
+  impact 1.0
+  title 'Disable content-type sniffing'
+  desc 'It prevents browser from trying to mime-sniff the content-type of a response away from the one being declared by the server. It reduces exposure to drive-by downloads and the risks of user uploaded content that, with clever naming, could be treated as a different content-type, like an executable.'
+  describe parse_config_file(nginx_hardening, options_add_header) do
+    its('add_header') { should include 'X-Content-Type-Options nosniff' }
+  end
+end
+
+control 'nginx-12' do
+  impact 1.0
+  title 'TLS Protocols'
+  desc 'When choosing a cipher during an SSLv3 or TLSv1 handshake, normally the client\'s preference is used. If this directive is enabled, the server\'s preference will be used instead.'
+  ref 'SSL Hardening config', url: 'https://mozilla.github.io/server-side-tls/ssl-config-generator/'
+  describe file(nginx_hardening) do
+    its('content') { should match(/^\s*ssl_protocols TLSv1.2;$/) }
+    its('content') { should match(/^\s*ssl_session_tickets off;$/) }
+    its('content') { should match(/^\s*ssl_ciphers 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256';$/) }
+    its('content') { should match(/^\s*ssl_prefer_server_ciphers on;$/) }
+    its('content') { should match(%r{^\s*ssl_dhparam /etc/nginx/dh2048.pem;$}) }
+    # its('content') { should match(/^\s*ssl on;$/) }
+  end
+end
+
+control 'nginx-13' do
+  impact 1.0
+  title 'Add HSTS Header'
+  desc 'HTTP Strict Transport Security (HSTS) is a web security policy mechanism which helps to protect websites against protocol downgrade attacks and cookie hijacking. It allows web servers to declare that web browsers (or other complying user agents) should only interact with it using secure HTTPS connections, and never via the insecure HTTP protocol. HSTS is an IETF standards track protocol and is specified in RFC 6797.'
+  describe file(nginx_hardening) do
+    its('content') { should match(/^\s*add_header Strict-Transport-Security max-age=15768000;$/) }
+  end
+end
+
+control 'nginx-14' do
+  impact 1.0
+  title 'Disable insecure HTTP-methods'
+  desc 'Disable insecure HTTP-methods and allow only necessary methods.'
+
+  describe file(nginx_conf) do
+    its('content') { should match(/^\s+if ($request_method !~ ^(GET|HEAD|POST)$ )$/) }
```

Comment:
We should reference https://www.owasp.org/index.php/Test_HTTP_Methods_(OTG-CONFIG-006) here. According to this doc it is optimized for plain file delivery via http. In such cases your configuration makes perfect sense. The other use case is an API behind nginx. In such a case you may need at least OPTIONS for CORS and PUT/POST/DELETE to manipulate the state.